### PR TITLE
Add batch score environment to azureml registry

### PR DESCRIPTION
**Pull request by yetamsft**
This environment is used in the batch-score component, so it is required to be published to azureml registry before the component.
Related component PR: https://github.com/Azure/azureml-assets/pull/1882

### CODE REVIEW

File: assets/batch_score/environments/context/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,17 @@
+FROM mcr.microsoft.com/azureml/openmpi4.1.0-ubuntu22.04
```
Comment (vizhur) [inline assets/batch_score/environments/context/Dockerfile:1]: use template by adding :{{latest-image-tag}}

see example: https://github.com/Azure/azureml-assets/blob/main/assets/responsibleai/environments/responsibleai-ubuntu20.04-py38-cpu/context/Dockerfile

File: assets/batch_score/environments/context/conda_dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: batch-score-env
+channels:
+  - defaults
+dependencies:
+  - python=3.8
+  - pip
+  - pip:
+      - mlflow
+      - azureml-core
```
Comment (vizhur) [inline assets/batch_score/environments/context/conda_dependencies.yaml:9]: consider pinning package versions. for reproducibility. we recommend to use a template to resolve the latest on pypi at least for azureml-packages 

=={{latest-pypi-version}}

File: assets/batch_score/environments/environment.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+image:
+  name: azureml/curated/batch-score-env
```
Comment (vizhur) [inline assets/batch_score/environments/environment.yaml:2]: remove '-env'

**Review by vizhur — CHANGES_REQUESTED** [review]
please address comments

File: assets/batch_score/environments/environment.yaml (hunk 3 above)
Comment (yetamsft) [reply assets/batch_score/environments/environment.yaml]: `batch-score-llm` sounds good to me. Renamed.

File: assets/batch_score/environments/context/conda_dependencies.yaml (hunk 2 above)
Comment (yetamsft) [reply assets/batch_score/environments/context/conda_dependencies.yaml]: Sounds good. Added.

File: assets/batch_score/environments/context/Dockerfile (hunk 1 above)
Comment (yetamsft) [reply assets/batch_score/environments/context/Dockerfile]: Added.

**Pull request closed without merging**